`zig fmt` fails with unhelpful error message when a comment contains a tab

Formatting the following code with `zig fmt` fails:

```zig
//	this is broken!
```

```
fail.zig:1:1: error: Expected test, comptime, var decl, or container field, found 'Invalid'
// this is broken!
```

It does not matter where the comment with the tab is, the error message stays the same, pointing to `filename.zig:1:1`, so it's hard to find what causes this

--- a/lib/std/json.zig
+++ b/lib/std/json.zig
@@ -886,6 +886,7 @@ pub const TokenStream = struct {
 
     pub fn next(self: *TokenStream) Error!?Token {
         if (self.token) |token| {
+            // TODO: Audit this pattern once #2915 is closed
             const copy = token;
             self.token = null;
             return copy;

--- a/lib/std/zig/tokenizer.zig
+++ b/lib/std/zig/tokenizer.zig
@@ -411,8 +411,9 @@ pub const Tokenizer = struct {
 
     pub fn next(self: *Tokenizer) Token {
         if (self.pending_invalid_token) |token| {
+            const copy = token;
             self.pending_invalid_token = null;
-            return token;
+            return copy;
         }
         const start_index = self.index;
         var state = State.Start;
@@ -1265,8 +1266,9 @@ pub const Tokenizer = struct {
 
         if (result.id == Token.Id.Eof) {
             if (self.pending_invalid_token) |token| {
+                const copy = token;
                 self.pending_invalid_token = null;
-                return token;
+                return copy;
             }
         }
 

--- a/lib/std/zig/tokenizer.zig
+++ b/lib/std/zig/tokenizer.zig
@@ -411,6 +411,7 @@ pub const Tokenizer = struct {
 
     pub fn next(self: *Tokenizer) Token {
         if (self.pending_invalid_token) |token| {
+            // TODO: Audit this pattern once #2915 is closed
             const copy = token;
             self.pending_invalid_token = null;
             return copy;
@@ -1266,6 +1267,7 @@ pub const Tokenizer = struct {
 
         if (result.id == Token.Id.Eof) {
             if (self.pending_invalid_token) |token| {
+                // TODO: Audit this pattern once #2915 is closed
                 const copy = token;
                 self.pending_invalid_token = null;
                 return copy;

